client/lobby/SelectionTab.cpp

```diff
@@ -328,29 +345,82 @@ void SelectionTab::showPopupWindow(const Point & cursorPosition)
 	if(py >= curItems.size())
 		return;
 
-	std::string text = boost::str(boost::format("{%1%}\r\n\r\n%2%:\r\n%3%") % curItems[py]->getName() % CGI->generaltexth->translate("vcmi.lobby.filename") % curItems[py]->fileURI);
-	if(curItems[py]->date != "")
-	    text += boost::str(boost::format("\r\n\r\n%1%:\r\n%2%") % CGI->generaltexth->translate("vcmi.lobby.creationDate") % curItems[py]->date);
+	if(!curItems[py]->isFolder)
+	{
+		std::string text = boost::str(boost::format("{%1%}\r\n\r\n%2%:\r\n%3%") % curItems[py]->getName() % CGI->generaltexth->translate("vcmi.lobby.filename") % curItems[py]->fileURI);
+		if(curItems[py]->date != "")
+			text += boost::str(boost::format("\r\n\r\n%1%:\r\n%2%") % CGI->generaltexth->translate("vcmi.lobby.creationDate") % curItems[py]->date);
 
-	CRClickPopup::createAndPush(text);
+		CRClickPopup::createAndPush(text);
+	}
+}
+
+std::tuple<std::string, std::string, bool, bool> SelectionTab::checkSubfolder(std::string path)
+{
+	std::string folderName = "";
+	bool parentExists = (curFolder != "");
+	bool fileInFolder = false;
+
+	std::vector<std::string> filetree;
+	// delete first element (e.g. 'MAPS')
+	boost::split(filetree, path, boost::is_any_of("/"));
+	filetree.erase(filetree.begin());
+	std::string pathWithoutPrefix = boost::algorithm::join(filetree, "/");
+
+	std::string baseFolder = boost::filesystem::path(pathWithoutPrefix).parent_path().string();
```

Comment:
these paths come from our virtual filesystem, not from actual system filesystem, better not to use boost::filesystem with it.
```suggestion
	filetree.pop_back();
	std::string baseFolder = boost::algorithm::join(filetree, "/");
```